controllers/listingAPI.js

```diff
@@ -12,144 +12,126 @@ const auth = require('../utils/authUtil');
-						baseView.renderWithAddons(req, res, 'pages/purchaseSummary', {result});
-					},
-				notFound: 
-					() => htmlResponse.fail(req, res, 'Could not get your purchase summary :(', 'Payment Summary Not Found'),
-			});
-		},
-		notFound: () => htmlResponse.fail(req, res, 'Could not get your session ID', 'Session Failure.'),
-	});
+    console.log('Received request to see purchase summary.')
+    auth.getSessionFromCookie(req, 
+        {
+        found: (session) => 
+        {
+            var userPK = session.SS_US;
+            listingModel.GetPurchaseSummary(req.params.purchaseID, userPK, 
+                {
+                found:
+                    (result) => 
+                    {
+                        console.log(result);
+                        // res.render(view.viewPurchaseSummary(result));
+                        console.log("hello my name is cameron");
+                    },
+                notFound:
+                    () => htmlResponse.fail(req, res, 'Could not get your purchase summary :(', 'Payment Summary Not Found'),
+            });
+        },
+        notFound: () => htmlResponse.fail(req, res, 'Could not get your session ID', 'Session Failure.'),
+    });
 });
 
 router.get('/purchase=:listingID&quantity=:amount', (req, res) => 
 {
-	console.log(req.params.listingID+ "|" + req.params.amount);
-	auth.getSessionFromCookie(req, 
-	{
-		found: (session) =>
-		{
-			var userPK = session.SS_US;
-			listingModel.getPrePurchaseInformation(userPK, req.params.listingID, req.params.amount,
-			{
-				success: (purchaseInfo) => 
-				{
-					//pass this onto the view.
-				},
-				fail: () => { }
-			});
-		},
-		notFound: () => { }
-	});
-});
+    console.log(req.params.listingID + "|" + req.params.amount);
+    auth.getSessionFromCookie(req, 
+        {
+        found: (session) => 
+        {
+            var userPK = session.SS_US;
 
-router.post('/purchaseItem', (req, res) => 
-{
-	console.log(req);
-	listingModel.purchaseItem(
-		req.params.listingPK, 
-		req.params.buyerPK, 
-		req.params.paymentMethodPK, 
-		req.params.deliveryAddressPK, 
-		totalPrice, 
-		quantity,
-	{
-		success: (purchasePK) =>
-		{
-			res.redirect("/summary=" + purchasePK);
-		},
-		fail: (reason) => { res.send(jsonResponse.fail(reason)); }
-	})
-});
+            listingModel.getPrePurchaseInformation(userPK, req.params.listingID, req.params.amount, 
+                {
+                success: (purchase) => 
+                {
+                    baseView.renderWithAddons(req, res, 'pages/purchase', { purchase });
+                    //pass this onto the view.
 
-router.get('/confirmPurchase', (req, res) =>
-{
-	let currentUser = req.cookies.currentUser;
-	console.log('currentUser is ' + currentUser);
-	baseView.renderWithAddons(req, res, 'pages/confirmPurchase', { currentUser })
+                },
+                fail: () => htmlResponse.fail(req, res, 'Could not get your purchase :(', 'Payment Not Found'),
```

Comment:
this isn't technically correct, either pass through the reason from 'getPrePurchaseInformation', or just say 'Error loading Purchase Screen', this isn't specifically the payment, its loading other things too